onnxruntime/core/providers/op_kernel_type_control.h

```diff
@@ -186,6 +186,40 @@ struct EnabledTypes {
                                             ::onnxruntime::op_kernel_type_control::kAllOpSets, \
                                             ArgDirection, ArgIndex, __VA_ARGS__)
 
+/**
+ * TypeList type with the supported types for a given Op kernel argument.
+ *
+ * @param OpProvider The Op provider.
+ * @param OpDomain The Op domain.
+ * @param OpName The Op name.
+ * @param OpSet The opset to use for the supported types list.
+ * @param ArgDirection Direction of the given Op kernel argument - Input or Output.
+ * @param ArgIndex Index of the given Op kernel argument.
+ */
+#define ORT_OP_KERNEL_ARG_SUPPORTED_TYPE_LIST(                                                          \
+    OpProvider, OpDomain, OpName, OpSet, ArgDirection, ArgIndex)                                        \
+  ::onnxruntime::op_kernel_type_control::TypesHolder<                                                   \
+      ::onnxruntime::op_kernel_type_control::tags::Supported<                                           \
+          ORT_OP_KERNEL_TYPE_CTRL_INTERNAL_OP_KERNEL_ARG_TAG(OpDomain, OpName, ArgDirection, ArgIndex), \
+          ::onnxruntime::op_kernel_type_control::                                                       \
+              ORT_OP_KERNEL_TYPE_CTRL_INTERNAL_PROVIDER_TAG_CLASS_NAME(OpProvider),                     \
+          OpSet>>::types
```

Comment:
could we put the Supported TypesHolder type into a separate macro (perhaps ORT_OP_KERNEL_TYPE_CTRL_INTERNAL_SUPPORTED_TYPES_HOLDER) and use that here and in ORT_OP_KERNEL_ARG_ENABLED_TYPE_LIST?